### DISCUSSION TITLE
Availability of updated data when querying calculated time series

### DISCUSSION THREAD

**Discussion by aes-powel** [Volue-Public/energy-mesh-python#172] [Q&A] [answered]
Let’s assume in Mesh there are two time series A and B where we can put values on. Time series C is then a derived calculation from A and B using a formula.

If we put values on A and/or B through the API, and then we read time series C, will we obtain the results of the calculation that includes the new values we just put on A and/or B? Or do we need to wait for a specific amount of time before querying?

**Comment by erny-powel** [comment]
If this is happening in the same session the result should include the new values. If it does not that is a bug.

**Comment by lvojeroen** [comment]
Can you clarify what the expected behavior is when this happens in different sessions? A new session gets the calculated values when querying for C as soon as the session that put values on A and/or B has committed its transaction?

**Reply by tnoczyns-volue** [reply]
> Can you clarify what the expected behavior is when this happens in different sessions? A new session gets the calculated values when querying for C as soon as the session that put values on A and/or B has committed its transaction?

Yes, that is the expected behavior. So as soon as a session will commit its changes, other sessions upon reading (e.g. on another `read_timeseries_points` call) will get updated values.

**Comment by lvojeroen** [comment, marked as answer]
Thank you for the clarifications!

**Reply by tnoczyns-volue** [reply]
Just had a discussion with @erny-powel. 

> So as soon as a session will commit its changes, other sessions upon reading (e.g. on another read_timeseries_points call) will get updated values.`

May not always be true. The procedure of updating other sessions is done asynchronously and may not complete before returning from `commit` RPC. So depending on the system load, scheduling, etc. other sessions may still have old values even after the `commit` was done.

I'll look into this and confirm, meanwhile please use some short timeout - chosen empirically.

**Reply by tnoczyns-volue** [reply]
The issue was fixed and will be available with next release of Mesh. The change was done on Mesh level, not the Python SDK. With the fix all sessions upon reading are guaranteed to get updated values like described before.

**Reply by lvojeroen** [reply]
Thanks for the update, that is good news. Just for clarity: if you say "next release of Mesh", is that the 2.4 release?

**Reply by tnoczyns-volue** [reply]
Yes, 2.4.

**Comment by tnoczyns-volue** [comment]
@aes-powel, @lvojeroen can we mark this discussion as answered? Or there are still some questions?

**Comment by lvojeroen** [comment]
For me, this is clear. I don't think I mark this discussion as answered since I didn't post the question, so I'll let @aes-powel do that.
